### DISCUSSION TITLE
Need help with my config 🙏🙏🙏

### DISCUSSION THREAD

**Discussion by iantubello** [maca88/SmartBikeLights#9] [Q&A] [answered]
Hi, so I am trying to configure a setting for my UT800 and RTL515. I am running 4 default activity profiles on my Head unit, Commute, Indoor, Mountain and Road.

It is a bit complex but I hope you can help me. I dont really understand the sequencing of the triggers but I hope what I did makes sense.

`1,1|All:1H]0#0,73413136#19,7|ComLow:3:4:1K=COMMUTE:H]1C[1.38889|Commut:3:3:1K=COMMUTE:H]1C]1.38889|MtbDay:3:0:1K=MOUNTAIN:H]1Er0,s-150|MtbNit:3:2:1K=MOUNTAIN:H]1Es-150,r0|RdDay:3:0:1K=ROAD:H=3Er0,s-150|RdNit:3:4:1K=ROAD:H=3Es-150,r0|:1:3:1D=1#0,73535488#34,11|ComWrn:4:6:1K=COMMUTE:H]1C]0I[140]0|ComDgr:4:7:1K=COMMUTE:H]1C]0I[140]1|CmEBrk:3:7:1.5K=COMMUTE:H]2A[-50|ComBrk:3:4:1K=COMMUTE:H]1A[-30|Commut:3:5:1K=COMMUTE:H]1C]0.27778|MTB:1:0:1K=MOUNTAIN:|RdWrn:4:6:1K=ROAD:H]1C]0I[50]0|RdBrk:3:4:1K=ROAD:H]1A[-30|RdDay:4:0:1K=ROAD:H]1Er0,s-150C]0.27778|RdNit:4:5:1K=ROAD:H]1Es-150,r0C]0.27778|:1:5:1D=1#7,5:UT800|2,:-1,Off:0|1,High:2|1,Medium:3|1,Low:4|2,Day Strobe:7,Night Pulse:6#6,3:RTL515|2,:-1,Off:0|2,High:4,Low:5|2,Day Strobe:7,Night Pulse:6#B3570#4#3#0#1 `

Updated config:

`1,1|All:1H]1#0,73413136#9,5|ComStp:2:4:1K=COMMUTE:C[1.38889|Commut:1:3:1K=COMMUTE:|MTB:1:2:1K=MOUNTAIN:|Road:4:4:1K=ROAD:H=3Es-150,r0C]0|:1:0:1D=1#0,73535488#21,9|ComDgr:3:7:1K=COMMUTE:C]0I[140]1|ComWrn:3:6:1K=COMMUTE:C]0I[140]0|RdWrn:3:6:1K=ROAD:C]0I[50]0|ComBrk:2:6:1K=COMMUTE:A[-4.16667|RdBrk:2:4:1K=ROAD:A[-4.16667|ComStp:2:6:1K=COMMUTE:C[1.38889|RdNit:3:5:1Es-150,r0K=ROAD:C]1.38889|Commut:2:4:1K=COMMUTE:C]0.27778|:1:0:1D=1#7,5:UT800|2,:-1,Off:0|1,High:2|1,Medium:3|1,Low:4|2,Day Strobe:7,Night Pulse:6#6,3:RTL515|2,:-1,Off:0|2,Solid:4,Peleton:5|2,Day Strobe:7,Night Pulse:6#B3570#4#3#0#1 `

**Comment by maca88** [comment, marked as answer]
Hi,

The configuration makes sense, but you could simplify it by doing the following:

1. As all filter groups have timer state `> ST` or higher, you can change the global filter to `> ST` and remove them in the group filters.
2. By setting the default mode to `Off` you can remove the filter groups that set them to `Off`
3. The filter group `ComDgr` should be before `ComWrn` as with the current configuration, in case you have multiple vehicles behind you with both `Medium` and `High` threats, the `ComWrn` would become active, which is probably not desired

Here is a simplified version:
`
1,1|All:1H]1#0,73413136#10,6|ComLow:2:4:1K=COMMUTE:C[1.38889|Commut:1:3:1K=COMMUTE:|MtbNit:2:2:1K=MOUNTAIN:Es-150,r0|RdNit:3:4:1K=ROAD:H=3Es-150,r0|RdDay:1:3:1K=ROAD:|:1:0:1D=1#0,73535488#21,9|ComDgr:3:7:1K=COMMUTE:C]0I[140]1|ComWrn:3:6:1K=COMMUTE:C]0I[140]0|CmEBrk:3:7:1.5K=COMMUTE:H]2A[-50|ComBrk:2:4:1K=COMMUTE:A[-30|Commut:2:5:1K=COMMUTE:C]0.27778|RdWrn:3:6:1K=ROAD:C]0I[50]0|RdBrk:2:4:1K=ROAD:A[-30|RdNit:2:5:1K=ROAD:Es-150,r0|:1:0:1D=1#7,5:UT800|2,:-1,Off:0|1,High:2|1,Medium:3|1,Low:4|2,Day Strobe:7,Night Pulse:6#6,3:RTL515|2,:-1,Off:0|2,High:4,Low:5|2,Day Strobe:7,Night Pulse:6#B3570#4#3#0#1
`

Maybe you can explain what you expect from each activity profile so that I can check whether the configuration matches that.

**Reply by iantubello** [reply]
Awesome!

I am so sorry if this is a long read and I hope it doesn't hurt your eyes. I tried to append with a _ as much as I could but I was just typing from a phone.

But thank you very much for your help snd creating this wonderful program!

So here are the profiles and the concept behind them:

My Head unit Activity profiles:
_COMMUTE - This is the profile that I would tend to run when cycling on a road that is shared by motor vehicles.
_INDOOR - In my other configurations I actually had filters to turn the lights off during this profile, but edge automatically disconnects from the lights when this profile is run and recording has started.
_MOUNTAIN - This is the profile I use for riding trails. There is no need for a taillight and the headlight can be run only when the sun sets. And before the sun rises.
_ROAD - we normally ride on cycling roads here where bright lights are unnecessary. So I actually want to use this exclusively for cycling roads. Lights should be turned off when the sun is up, except when Braking.

Brake lights:
So I had an idea to have brake lights glow up when decelerating and staying bright when stopped. I also had this idea to use emergency brake lights. I got the idea from when cars would suddenly decelerate and their taillights would flash fast.

Headlight:
_ComLow - for the activity Profile "COMMUTE", this trigger just sets the headlight to the lowest setting when stopped, like on a traffic light.
_Commut - these are the regular running lights for the profile "COMMUTE", should be triggered when the bike accelerates from a complete stop. Like when the traffic light turns green. The light should turn brighter one lever whne the bike starts to roll away.
_MtbDay - this was just trigger to tell the system to turn off the light when the profile "MOUNTAIN" is used and when the sun is up. Just to make sure that the lights are off during the day in case I have the headlights on my bike, helmet or on a pack nearby. This was ommited.
_MtbNit - this is a trigger to tell the system to turn on the lights on full brightness when the sun sets, when the profile "MOUNTAIN" is in use. As in trail mode.
_RdNit - this is a trigger to tell the system to put the headlight on the lowest setting when the profile "ROAD" is used. This is to be triggered when the sun sets.
_RdDay - this was just a trigger to tell the system to turn off the lights during the day when the sun is up when the "ROAD" profile is used. This was ommited.

I don't have a deep understanding of how the radar works yet but I hope the following vehicle detections and threats would be similar to how Garmin programmed the radar. Further study might be needed.

Taillight:
_ComDgr - this is a radar trigger that tells the light to blink fast  when the profile is "COMMUTE" and when a vehicle <140m away with a high threat level appears.
_ComWrn - this is a radar trigger that tells the the light to pulsate when the profile being used is "COMMUTE" and when a vehicle <140m away with a threat level of medium appears.
_ComEBrk - this is a brake light trigger that tells the system to blink fast when profile "COMMUTE" is used and when a sudden rapid deceleration occurs. For those tire skidding moments 😅
_ComBrk - this a brake light that tells the system to assign the tail light to night pulsing when the profile "COMMUTE" is used and when the bike decelerates or when the bike is in a complete stop. For those regular and normal braking.
_Commut - this is the trigger that normally runs when the profile "COMMUTE" is used and when the speed is >1Kph. Light mode bas been changed to solid.
_MTB - This is a trigger that tells the system to turn off the taillight when the profile "MOUNTAIN" is running. This was ommited.
_RdWrn - this a trigger that normally runs when the profile "ROAD" is running and will tell the system to switch the lights to night pulsing when a vehicle (or in this case a faster rider) is detected <100m with a threat level of >none.
_RdBrk - this is a trigger that runs when the profile "ROAD" is selected and a deceleration is detected.
_RdNit - This is the normal running taillight (mode is peleton) when the profile "ROAD" is running, during the night and when the speed is >5kph.

So I've changed the running (from peleton to solid) and brake lights (from solid to night pulsing) of the commute profile just to make things much more safer.

I hope the changes I did does not trigger an error. If they do please let me know.

`1,1|All:1H]1#0,73413136#9,5|ComLow:2:4:1K=COMMUTE:C[1.38889|Commut:1:3:1K=COMMUTE:|MtbNit:2:2:1K=MOUNTAIN:Es-150,r0|RdNit:3:4:1K=ROAD:H=3Es-150,r0|:1:0:1D=1#0,73535488#22,9|ComDgr:3:7:1K=COMMUTE:C]0I[140]1|ComWrn:3:6:1K=COMMUTE:C]0I[140]0|CmEBrk:3:7:1.5K=COMMUTE:H]2A[-50|ComBrk:2:6:1K=COMMUTE:A[-30|Commut:2:4:1K=COMMUTE:C]0.27778|RdWrn:3:6:1K=ROAD:C]0I[50]0|RdBrk:2:4:1K=ROAD:A[-30|RdNit:3:5:1K=ROAD:Es-150,r0C]1.38889|:1:0:1D=1#7,5:UT800|2,:-1,Off:0|1,High:2|1,Medium:3|1,Low:4|2,Day Strobe:7,Night Pulse:6#6,3:RTL515|2,:-1,Off:0|2,Solid:4,Peleton:5|2,Day Strobe:7,Night Pulse:6#B3570#4#3#0#1`

**Reply by iantubello** [reply]
Hi,
So I’ve simplified it a lot more and came up with a new configuration.

`1,1|All:1H]1#0,73413136#8,4|ComLow:2:4:1K=COMMUTE:C[1.38889|Commut:1:3:1K=COMMUTE:|RdNit:4:4:1K=ROAD:H=3Es-150,r0C]0|:1:0:1D=1#0,73535488#21,9|ComDgr:3:7:1K=COMMUTE:C]0I[140]1|ComWrn:3:6:1K=COMMUTE:C]0I[140]0|RdWrn:3:6:1K=ROAD:C]0I[50]0|ComBrk:2:6:1K=COMMUTE:A[-4.16667|RdBrk:2:4:1K=ROAD:C[-4.16667|ComStp:2:6:1K=COMMUTE:C[1.38889|RdNit:3:5:1Es-150,r0K=ROAD:C]1.38889|Commut:2:4:1K=COMMUTE:C]0.27778|:1:0:1D=1#7,5:UT800|2,:-1,Off:0|1,High:2|1,Medium:3|1,Low:4|2,Day Strobe:7,Night Pulse:6#6,3:RTL515|2,:-1,Off:0|2,Solid:4,Peleton:5|2,Day Strobe:7,Night Pulse:6#B3570#4#3#0#1 `

When I spun the bike on an indoor trainer with no resistance my lights were still off on the profile ‘ROAD’, even when the light mode were on smart. Is it a problem with how Garmin does the road profile where it actually looks for an input from the speed sensor and a movement from GPS? Might have to investigate this.

I went into the simulation mode of the varia and found that it has another light mode when an incoming vehicle has a high threat and is near. It blinks even faster. Is there a way to get this on the filters?

Thanks!

**Reply by maca88** [reply]
> I don't have a deep understanding of how the radar works yet but I hope the following vehicle detections and threats would be similar to how Garmin programmed the radar.

The radar filter uses the [CIQ BikeRadar API](https://developer.garmin.com/connect-iq/api-docs/Toybox/AntPlus/BikeRadar.html), which gets the data directly from the Varia radar.

> no resistance my lights were still off on the profile ‘ROAD’, even when the light mode were on smart. 

Acceleration, Speed and Sunset/Sunrise filters require GPS signal in order to work.

> Is it a problem with how Garmin does the road profile where it actually looks for an input from the speed sensor and a movement from GPS?

The problem is that [currentSpeed](https://developer.garmin.com/connect-iq/api-docs/Toybox/Activity/Info.html#currentSpeed-var) requires GPS signal to work.

> So I’ve simplified it a lot more and came up with a new configuration.

With the last configuration that you posted I spotted two things:
1. Headlight filter group `MtbNit` is missing
2. Taillight filter group `RdBrk` has `Speed` filter instead of `Acceleration`

other than that the configuration is fine.

> I went into the simulation mode of the varia and found that it has another light mode when an incoming vehicle has a high threat and is near. It blinks even faster. Is there a way to get this on the filters?

Unfortunately not, that mode is not accessible via the [BikeLight API modes](https://developer.garmin.com/connect-iq/api-docs/Toybox/AntPlus/BikeLight.html#getCapableModes-instance_function).

**Reply by iantubello** [reply]
Hi, So I’ve been testing the config outside for a couple of days now. At first, the headlight was still all over the place and switching to high when going faster. Switched the setting from auto to individual within the garmin widget and voila! Everything is smooth as butter now!

I can comment on the light mode I was referring to when a high threat vehicle is approaching. And it seems that the radar switches to that mode automatically, so it’s all good!

i went back and put the MTB headlight filter back in.


If anyone wants the config here it is:

`1,1|All:1H]1#0,73413136#9,5|ComStp:2:4:1K=COMMUTE:C[1.38889|Commut:1:3:1K=COMMUTE:|MTB:1:2:1K=MOUNTAIN:|Road:4:4:1K=ROAD:H=3Es-150,r0C]0|:1:0:1D=1#0,73535488#21,9|ComDgr:3:7:1K=COMMUTE:C]0I[140]1|ComWrn:3:6:1K=COMMUTE:C]0I[140]0|RdWrn:3:6:1K=ROAD:C]0I[50]0|ComBrk:2:6:1K=COMMUTE:A[-4.16667|RdBrk:2:4:1K=ROAD:A[-4.16667|ComStp:2:6:1K=COMMUTE:C[1.38889|RdNit:3:5:1Es-150,r0K=ROAD:C]1.38889|Commut:2:4:1K=COMMUTE:C]0.27778|:1:0:1D=1#7,5:UT800|2,:-1,Off:0|1,High:2|1,Medium:3|1,Low:4|2,Day Strobe:7,Night Pulse:6#6,3:RTL515|2,:-1,Off:0|2,Solid:4,Peleton:5|2,Day Strobe:7,Night Pulse:6#B3570#4#3#0#1 `

The program is running extremely well and solves a lot of problems the garmin light system has. Garmin should hire maca88 to help get rid of the nasty bugs garmin pumps out.

Thank you for your hard word maca88!
